packages/@ember/-internals/utils/lib/trackable-object.ts

```diff
@@ -0,0 +1,12 @@
+import symbol from './symbol';
+
+export const TRACKABLE_OBJECT = symbol('TRACKABLE_OBJECT');
+
+export function isTrackableObject(obj: any) {
```

Comment:
Two quick TS notes on this particular implementation:

1. Now that it's available, we strongly prefer `unknown` to `any` for scenarios like this. (There's almost nowhere I'd use `any` at this point, in fact.)
2. As a matter of good hygiene, we should explicitly annotate the return type of this (and indeed all functions). While TS can infer it, expressing our intent that way makes it obvious if we accidentally violate our intent.